### DISCUSSION TITLE
In the Red

### DISCUSSION THREAD

**Discussion by mcfnord** [jamulussoftware/jamulus#2968] [Documentation]
Our docs say [red indicates the delay is too large for jamming](https://jamulus.io/wiki/Software-Manual). It isn't, though higher delays can require certain adjustments, techniques, and new skills. Today I overheard a new user read somewhere that red delay indicated no fun likely. (Can't locate what document she was reading from.) We say red means no jamming, pool's closed, but that's so untrue. Jitter is more fatal than delay. I don't want to discourage people from doing exactly as I do, which appears to be outside the expected or typical usage of Jamulus, because my delay is often cherry red, yet I'm jamming. There are major musical and cultural opportunities when you manage delay successfully. We're pitching as the e-Studio in your City, but we're actually the Global HAM Radio for Music.

I wonder what we'd write in a Jamulus High Delay Handbook.

**Comment by ann0see** [comment]
This highly depends on your skill, musicality and style of music. The color coding is for user convenience. There's always a tradeoff between 100% correct and widely understandable. That's why people should listen and look at the delay stats. Of course, we can soften the meaning of the red light. 

My choir leader (who's also an Organist and is used to high latency) can play fine with > 100ms delay over 5G, people do jams spread over America and Europe. Others complain about 30-50ms being way too much delay and give up early.

If you plan to write something up for high latency jamming, you can use the community knowledge base.

**Comment by dcorson-ticino-com** [comment]
Yes, in general the green (<43ms), yellow (<68ms) and red (more) are well placed.
Of course with certain pieces and with experience more is possible.  I play regularly chamber music across the Atlantic which gives total delays of around 100ms and it works, just.  We play every week now for 2 years now so we are used to it, but still, for the music we play, delays of more than 120ms make playing together more or less impossible. 
If we could get down to 60ms or less it would be much more relaxed and fun.  It's just that physics makes that that is not going to happen.

**Comment by pcar75** [comment]
Yep, barring quantum instant communication, it depends greatly on the individuals tolerance and adaptability to delay.  But it also depends on the material : some fast boogie woogie may sound like a train wreck with "reder delayed" contributors.  
As extreme as it may seems, I know this one guy , well known -and liked- in public Jamulus sessions in the Americas and Europe with really big delay.  Sometimes, he uses the Jamulus return in one ear and his direct monitor in the other and , as he says, he sings right in between.  It works surprisingly well many times, so my hat to this genius.
Couldn't resist giving my 0.02$ :)))

**Comment by gilgongo** [comment]
The observation about what we say about the red light (which, BTW we could have as just "Might make it too hard to play"?) also relates to the wider topic of "telemetry" in general.  Despite the presence of the Ping, Jitter and Delay lights, there isn't actually any reliable formula for what to do if, say, the Delay goes red or the Ping spikes up. 

It's possible to do some experimentation with manual jitter buffer settings (a trade-off between audio quality and overall delay), but the effect of these LEDs is really just to show that "things are happening"  :-)  As @ann0see says, it's your ears that matter when it comes to deciding whether you can play with others.

**Comment by ann0see** [comment]
Locked due to spam. Sorry @mcfnord 

Edit: unlocked.
